src/hotspot/share/opto/addnode.cpp

```diff
@@ -235,6 +235,45 @@ const Type *AddNode::add_of_identity( const Type *t1, const Type *t2 ) const {
   return NULL;
 }
 
+//---------------------------Bitmask helper------------------------------------
+// Get a bitmask for expression: "(a & 0xFF) << 8" -> 0xFF00
+static juint get_bitmask(PhaseGVN *phase, Node* value) {
+  int opcode = value->Opcode();
+  if (opcode == Op_LShiftI && value->in(2)->is_Con()) {
+    int lshift = phase->type(value->in(2))->is_int()->get_con();
+    return get_bitmask(phase, value->in(1)) << lshift;
+  } else if (opcode == Op_AndI && value->in(2)->is_Con()) {
+    return phase->type(value->in(2))->is_int()->get_con();
+  } else if (opcode == Op_BitfieldInsertI) {
+    int width = value->in(3)->get_int();
+    int offset = value->in(4)->get_int();
+    int mask = ((1 << width) - 1) << offset;
+    return mask | get_bitmask(phase, value->in(1));
+  } else if (opcode == Op_OrI) {
+    return get_bitmask(phase, value->in(1)) |
+           get_bitmask(phase, value->in(2));
+  }
+  return 0xFFFFFFFF;
+}
+
+static julong get_bitmaskL(PhaseGVN *phase, Node* value) {
+  int opcode = value->Opcode();
```

Comment:
If you are not going to provide an equivalent comment to explain this routine then at least point the reader back to the comment for the preceding routine.